superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -154,67 +160,44 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
     }
   };
 
-  useEffect(() => {
-    const firstItem: SelectValue = data[0]
-      ? (groupby.map(col => data[0][col]) as string[])
-      : null;
-    if (!isDisabled && defaultToFirstItem && firstItem) {
-      // initialize to first value if set to default to first item
-      setValues(firstItem);
-    } else if (!isDisabled && defaultValue?.length) {
-      // initialize to saved value
-      setValues(defaultValue);
-    }
-  }, [defaultToFirstItem, defaultValue]);
-
```

Comment:
This `useEffect` has been combined with the other one that was checking for changes in controls (e.g. inverse selection etc). Now there are only two `useEffects` left - one to check for changes in data mask, and another to check for changes in control values.